### ISSUE TITLE
Android Drop-In:  No Routes Returned on some requests.... Back-end Issue I think.

### ISSUE THREAD

**Issue by EricGeiler** [mapbox/mapbox-navigation-android#6615]
Support Ticket # 134349

Drop-In UI: 2.9.0-rc.2
Android Platform

Example co-ordinates.

-79.444202,44.061903;-79.3735412970994,43.7947121000824

the above lat/long pairs return a route for cars or any vehicle for that matter as long as max_weight<=3
If we use max_weight=4 on the request, it fails to return a route... looking at tile query data for Mapbox I'm not seeing a reason for no routes.

https://api.mapbox.com/directions/v5/mapbox/driving-traffic/-79.444202,44.061903;-79.3735412970994,43.7947121000824?steps=true&alternatives=true&banner_instructions=true&roundabout_exits=true&voice_instructions=true&voice_units=metric&geometries=polyline6&language=en&overview=full&max_height=4.1&max_width=2&max_weight=3&exclude=toll%2Cferry&access_token=**REDACTED**

The above Request returns a set of directions based on the values provided, this is correct.

However the below request returns nothing (only change is max_weight=4 in the below req, versus max_weight=3 on the above.

https://api.mapbox.com/directions/v5/mapbox/driving-traffic/-79.444202,44.061903;-79.3735412970994,43.7947121000824?steps=true&alternatives=true&banner_instructions=true&roundabout_exits=true&voice_instructions=true&voice_units=metric&geometries=polyline6&language=en&overview=full&max_height=4.1&max_width=2&max_weight=4&exclude=toll%2Cferry&access_token=**REDACTED**

Here is the legal definition of HGV or Heavy Truck as it applies to the Province of Ontario.

**"Heavy Truck" definition: a commercial motor vehicle with a weight when unloaded, of three tonnes or more, or when loaded, of five tonnes or more, but does not include a passenger vehicle, an ambulance or any Vehicle of a police or fire department.

Local deliveries and collections are exempt from heavy truck restrictions if the location cannot be reached by any other road and provided that the route taken is the shortest possible to and from the location on the truck restricted route.

Failure to adhere to heavy truck restrictions may result in fines under the Highway Traffic Act and Municipal By-law.**

### Potential Reason for the results from a mapbox dev on slack.

“I found a thing in our internal fork of Valhalla that equates hgv=no to maxweight=3500 (w/ overrides in some countries like the U.S. but not Canada), unless there’s an explicit maxweight to override it.   I suppose hgv=destination would address the issue, but if 3.5 t isn’t the right cutoff, I’d flag it to support to get that changed in Canada.”

“Got it. I think we instituted a similar exception for access=private toward the endpoints of a route. I don’t know if that applies to the hgv=no thing, but if it does, maybe one issue is that the hgv=no is tagged only on Cummer Ave. but no further than that.”

blue circle = destination
black circles = showing hgv=no tags present on approach roads to this sub division.

![image](https://user-images.githubusercontent.com/61550186/202213315-c6163b5b-2d0d-4d94-9bb1-686aa94d2815.png)

### Expected behavior

A route should be generated.

### Actual behavior

No routes returned.

###  Summary

Basically if a truck in Ontario is completing a collection or delivery, they are allowed to enter via the shortest path, regardless of road restrictions for hgv= tag…. We would like to allow our drivers to complete pickup and deliveries in residentail areas, however currently we are seeing many directions request fail for LEGALLY allowed routes in Ontario, as soon as the weight exceeds 3.5 its seems.. I guess this lines up with the info Minh provided.

What do we have to do to make this work?  I really don’t want to just re-submit a directions req without using a weight value as its just a bandaid fix to the issue at hand.

**Comment by EricGeiler** [comment]
-79.44420; 44.061903;-79.4408506618072,43.8725524041208

Another example of a location that faces same type of routing issue.
